// @react-login-page/page8/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {};
exports["default"] = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _reactLoginPage = require("react-login-page");
Object.keys(_reactLoginPage).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _reactLoginPage[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _reactLoginPage[key];
    }
  });
});
var _Username = require("./control/login/Username");
Object.keys(_Username).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Username[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Username[key];
    }
  });
});
var _Password = require("./control/login/Password");
Object.keys(_Password).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Password[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Password[key];
    }
  });
});
var _Submit = require("./control/login/Submit");
Object.keys(_Submit).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Submit[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Submit[key];
    }
  });
});
var _Footer = require("./control/Footer");
Object.keys(_Footer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Footer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Footer[key];
    }
  });
});
var _Logo = require("./control/Logo");
Object.keys(_Logo).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Logo[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Logo[key];
    }
  });
});
var _Title = require("./control/Title");
Object.keys(_Title).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Title[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Title[key];
    }
  });
});
var _TitleLogin = require("./control/TitleLogin");
Object.keys(_TitleLogin).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _TitleLogin[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TitleLogin[key];
    }
  });
});
var _TitleSignup = require("./control/TitleSignup");
Object.keys(_TitleSignup).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _TitleSignup[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TitleSignup[key];
    }
  });
});
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["blocks", "extra", "data"],
  _excluded2 = ["name"],
  _excluded3 = ["name"],
  _excluded4 = ["children", "className"];
var RenderLogin = function RenderLogin() {
  var _useStore = (0, _reactLoginPage.useStore)(),
    _useStore$blocks = _useStore.blocks,
    blocks = _useStore$blocks === void 0 ? {} : _useStore$blocks,
    _useStore$extra = _useStore.extra,
    extra = _useStore$extra === void 0 ? {} : _useStore$extra,
    data = _useStore.data,
    label = (0, _objectWithoutProperties2["default"])(_useStore, _excluded);
  var _ref = data || {
      fields: []
    },
    fields = _ref.fields,
    buttons = _ref.buttons;
  var _useState = (0, _react.useState)('login'),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    panel = _useState2[0],
    setPanel = _useState2[1];
  var loginButtons = buttons.filter(function (m) {
    return m.name.indexOf("$$login") > -1;
  }).sort(function (a, b) {
    return a.index - b.index;
  });
  var loginFields = fields.filter(function (m) {
    return m.name.indexOf("$$login") > -1;
  }).sort(function (a, b) {
    return a.index - b.index;
  });
  var signupButtons = buttons.filter(function (m) {
    return m.name.indexOf("$$signup") > -1;
  }).sort(function (a, b) {
    return a.index - b.index;
  });
  var signupFields = fields.filter(function (m) {
    return m.name.indexOf("$$signup") > -1;
  }).sort(function (a, b) {
    return a.index - b.index;
  });
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Render, {
    children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: "login-page8-inner",
      children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("header", {
        children: [blocks.logo, blocks.title]
      }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("main", {
        children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("article", {
          className: panel === 'login' ? 'active' : '',
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("button", {
            onClick: function onClick() {
              return setPanel('login');
            },
            children: blocks['title-login']
          }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("section", {
            className: "login-page8-fields login-page8-login",
            children: [loginFields.map(function (item, idx) {
              var extraLabel = extra[item.name];
              if (!item.children && !extraLabel) return null;
              if (!item.children && extraLabel) return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
                children: extraLabel
              }, idx);
              if (!item.children) return null;
              var labelElement = label[item.name];
              var htmlFor = item.name.replace('$$login', '');
              var _item$children$props = item.children.props,
                name = _item$children$props.name,
                props = (0, _objectWithoutProperties2["default"])(_item$children$props, _excluded2);
              return /*#__PURE__*/(0, _jsxRuntime.jsxs)("label", {
                className: "rlp-".concat(item.name),
                htmlFor: htmlFor,
                children: [labelElement && /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
                  className: "login-page8-label",
                  children: labelElement
                }), /*#__PURE__*/(0, _react.cloneElement)(item.children, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, props), {}, {
                  name: panel === 'login' ? name : '',
                  key: item.name + idx
                })), extraLabel && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
                  children: extraLabel
                })]
              }, item.name + idx);
            }), /*#__PURE__*/(0, _jsxRuntime.jsx)("section", {
              className: "login-page8-button",
              children: loginButtons.map(function (item, idx) {
                var child = item.children;
                if (! /*#__PURE__*/(0, _react.isValidElement)(child)) return null;
                return /*#__PURE__*/(0, _react.cloneElement)(child, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props), {}, {
                  key: item.name + idx
                }));
              })
            })]
          })]
        }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("article", {
          className: panel === 'signup' ? 'active' : '',
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("button", {
            onClick: function onClick() {
              return setPanel('signup');
            },
            children: blocks['title-signup']
          }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("section", {
            className: "login-page8-fields login-page8-signup",
            children: [signupFields.map(function (item, idx) {
              var extraLabel = extra[item.name];
              if (!item.children && !extraLabel) return null;
              if (!item.children && extraLabel) return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
                children: extraLabel
              }, idx);
              if (!item.children) return null;
              var labelElement = label[item.name];
              var htmlFor = item.name.replace('$$signup', '');
              var _item$children$props2 = item.children.props,
                name = _item$children$props2.name,
                props = (0, _objectWithoutProperties2["default"])(_item$children$props2, _excluded3);
              return /*#__PURE__*/(0, _jsxRuntime.jsxs)("label", {
                htmlFor: htmlFor,
                children: [labelElement && /*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
                  className: "login-page8-label",
                  children: labelElement
                }), /*#__PURE__*/(0, _react.cloneElement)(item.children, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, props), {}, {
                  name: panel === 'signup' ? name : '',
                  key: item.name + idx
                })), extraLabel && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
                  children: extraLabel
                })]
              }, item.name + idx);
            }), /*#__PURE__*/(0, _jsxRuntime.jsx)("section", {
              className: "login-page8-button",
              children: signupButtons.map(function (item, idx) {
                var child = item.children;
                if (! /*#__PURE__*/(0, _react.isValidElement)(child)) return null;
                return /*#__PURE__*/(0, _react.cloneElement)(child, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props), {}, {
                  key: item.name + idx
                }));
              })
            })]
          })]
        })]
      }), blocks.footer]
    })
  });
};
var LoginPage = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var children = props.children,
    className = props.className,
    divProps = (0, _objectWithoutProperties2["default"])(props, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactLoginPage.Provider, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_Title.Title, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_TitleSignup.TitleSignup, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_TitleLogin.TitleLogin, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Logo.Logo, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Username.Username, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Password.Password, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Submit.Submit, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Username.Username, {
      panel: "signup",
      label: "E-mail",
      type: "email",
      placeholder: "E-mail",
      keyname: "e-mail"
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Password.Password, {
      panel: "signup",
      label: "Password",
      placeholder: "Password",
      keyname: "password"
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Password.Password, {
      panel: "signup",
      label: "Confirm Password",
      placeholder: "Confirm Password",
      keyname: "confirm-password"
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Submit.Submit, {
      keyname: "signup-submit",
      panel: "signup",
      children: "Signup"
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactLoginPage.Container, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, divProps), {}, {
      className: "login-page8 ".concat(className || ''),
      children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(RenderLogin, {}), children]
    }))]
  });
});
var Login = LoginPage;
Login.Username = _Username.Username;
Login.Password = _Password.Password;
Login.Submit = _Submit.Submit;
Login.Footer = _Footer.Footer;
Login.Logo = _Logo.Logo;
Login.Title = _Title.Title;
Login.TitleLogin = _TitleLogin.TitleLogin;
Login.TitleSignup = _TitleSignup.TitleSignup;
Login.displayName = 'BaseLoginPage';
var _default = exports["default"] = Login;
